### DISCUSSION TITLE
Berkeley Release ae78a64

### DISCUSSION THREAD

**Discussion by deepthiskumar** [MinaProtocol/mina#12757] [Berkeley]
Mina's ecosystem partner, O(1) Labs, is continuously working on improvements to the core protocol and today we are excited to announce an updated unstable release ae78a64 that is ready to test on the Berkeley testnet! Please report any issues in the comments on this post or tag any issues you create with berkeley-rampup and include berkeley ae78a64 in the issue title.

Improvements/features:
* Added a new field `implicit_account_creation_fee` to account updates, which can be used (on non-token parties) to subtract the creation fee from the transferred amount. https://github.com/MinaProtocol/mina/pull/10842
* Improved staking ledger sync during bootstrap https://github.com/MinaProtocol/mina/pull/11971
* Fix to prevent transaction pool DDOS https://github.com/MinaProtocol/mina/pull/12548
* Added an `access` permission to accounts https://github.com/MinaProtocol/mina/pull/12003
* Removed redundant fields `timestamp` and `global_slot_since_hard_fork` from zkApp preconditions https://github.com/MinaProtocol/mina/pull/12234
* Removed `transaction_expiry_hr` used for evicting long pending transaction from pool https://github.com/MinaProtocol/mina/pull/12231
* Added `valid_while` precondition to zkapp account updates https://github.com/MinaProtocol/mina/pull/12353
* Limit length of zkapp URIs to 255 characters https://github.com/MinaProtocol/mina/pull/12346
* Renames sequence events to actions https://github.com/MinaProtocol/mina/pull/12367
* Allow verification key update and usage in one zkapp transaction https://github.com/MinaProtocol/mina/pull/12398
* Add tokens from failed txns to archive DB https://github.com/MinaProtocol/mina/pull/12435
* Fix for https://github.com/MinaProtocol/mina/issues/12286 https://github.com/MinaProtocol/mina/pull/12450
* Ignore zkApp proofs for failed transactions during block validation https://github.com/MinaProtocol/mina/pull/12490
* Updated token call logic https://github.com/MinaProtocol/mina/pull/12512
* Specify verification key hash for proof updates https://github.com/MinaProtocol/mina/pull/12380
* New account permission to set vesting schedules https://github.com/MinaProtocol/mina/pull/12492
* Remove supercharged rewards https://github.com/MinaProtocol/mina/pull/12587
* Transaction hash prefix changed https://github.com/MinaProtocol/mina/pull/12651
* Fix for bootstrap failure https://github.com/MinaProtocol/mina/pull/12752
* Fix for transaction snark failure for zkapp transactions without any account updates https://github.com/MinaProtocol/mina/pull/12754

Please refer to the schema from the released commit

https://github.com/MinaProtocol/mina/blob/ae78a64a1ede801cbab748325566135f046993ff/src/app/archive/create_schema.sql
https://github.com/MinaProtocol/mina/blob/ae78a64a1ede801cbab748325566135f046993ff/src/app/archive/zkapp_tables.sql
Note: zkApps Test Transaction CLI tool cannot be used to create token accounts in this release.

Complete Changelog since 05c2f73 in:
Github Compare: https://github.com/MinaProtocol/mina/compare/05c2f73d0f6e4f1341286843814ce02dcb3919e0...ae78a64a1ede801cbab748325566135f046993ff

SnarkyJS change log: https://github.com/o1-labs/snarkyjs/blob/main/CHANGELOG.md
Upgrading & Connecting to Berkeley Testnet
Debian Packages:
Debian Repository:
First, set up and update the unstable Debian Repository for your platform. Replace the word CODENAME with the appropriate codename for your machine, one of bionic, focal, stretch, buster, or bullseye and run:

echo "deb [trusted=yes] http://packages.o1test.net/ CODENAME unstable" | sudo tee /etc/apt/sources.list.d/mina-unstable.list
sudo apt-get update
Then, install the package(s) that you need:
Mina Daemon: sudo apt-get install -y mina-berkeley=2.0.0rampup1-rampup-ae78a64
zkApps Test Transaction CLI: sudo apt-get install -y mina-zkapp-test-transaction=2.0.0rampup1-rampup-ae78a64
[zkApps Transaction CLI Documentation](https://github.com/MinaProtocol/mina/blob/ecbb3d490dce0f199894ebdf787a03b8fec64bd4/src/app/zkapp_test_transaction/README.md)

Docker Images:
Every docker image is built for each of the supported platforms, replace CODENAME in the images below with your choice of bionic, focal, stretch, buster, or bullseye to select the base docker image.
Mina Daemon: minaprotocol/mina-daemon:2.0.0rampup1-rampup-ae78a64-CODENAME-berkeley

Step by Step Guide:
This version will only connect and function properly on the newly launched Berkeley Testnet.

Create a libp2p keypair for a node the first time and persist it
mina libp2p generate-keypair -privkey-path <path-to-the-key-file>

Run mina daemon

Set environment variable MINA_LIBP2P_PASS with the password set for the libp2p keypair in step1
Make sure to run the daemon with -libp2p-keypair <path-to-the-key-file> and --peer-list-url [https://storage.googleapis.com/seed-lists/berkeley_seeds.txt]().
If you are running the correct version on the correct network BERKELEY TESTNET, mina client status will show:

Chain id: 8fab56ef04619087270631e89bde75990d7968309f4008e3cac42eee9601fc96
Git SHA-1:  ae78a64a1ede801cbab748325566135f046993ff

**Comment by duccas** [comment]
Docker images are not pushed to Docker hub. There is no way to pull.
```minaprotocol/mina-daemon:2.0.0rampup1-rampup-ae78a64-bionic-berkeley```

![image](https://user-images.githubusercontent.com/64695363/222881988-c1f1fa46-dcbd-46ad-8a89-b76dd2f1a0e8.png)

**Comment by psteckler** [comment]
`docker pull gcr.io/o1labs-192920/mina-daemon:ae78a64-bionic-berkeley`
